tendrl/commons/objects/cluster/atoms/delete_cluster_details/__init__.py

```diff
@@ -83,6 +84,31 @@ def run(self):
             # No cluster alerts, continue
             pass
 
+        # Remove cluster indexes for down node
+        try:
+            index_key = "/indexes/tags/tendrl/integration/%s" % integration_id
+            _node_ids = etcd_utils.read(
+                index_key
+            ).value
+            _node_ids = json.loads(_node_ids)
+            _node_arr = []
+            for _node_id in _node_ids:
+                node_obj = NS.tendrl.objects.NodeContext(
+                    node_id=_node_id
+                ).load()
+                if node_obj.status.lower() == "up":
```

Comment:
I would say if the node is down remove the node_id from list directly. Finally write the updated list to the etcd.

Reply:
ack